app/src/main/java/com/example/triptracker/viewmodel/MapPopupViewModel.kt

```diff
@@ -0,0 +1,18 @@
+package com.example.triptracker.viewmodel
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.example.triptracker.model.geocoder.NominatimApi
+
+open class MapPopupViewModel : ViewModel() {
+
+  private val _address = MutableLiveData<String>()
+  val address: LiveData<String> = _address
+
+  private val geocoder = NominatimApi()
+
+  fun fetchAddressForPin(latitude: Float, longitude: Float) {
```

Comment:
This creates a new instance everytime of the viewmodel ?

Reply:
No, that just calls the Nominatim API, the creation of the instance is in the actual MapPopups.kt class